front/src/modules/rollingStock/components/RollingStockEditor/CurveSpreadsheet.tsx

```diff
@@ -77,30 +80,69 @@ const CurveSpreadsheet = ({
         ...(isDefaultCurve ? { default_curve: formattedCurve } : {}),
       },
     };
+
+    // Set the updated effort curves
     setEffortCurves(updatedEffortCurve);
   };
 
-  const orderSpreadsheetValues = () => {
-    const orderedValuesByVelocity = spreadsheetCurve.sort((a, b) => {
-      // if a row has a max_effort, but no speed, it should appear at the top of the table
-      if (b[0] && b[0].value === '') return 1;
-      return Number(a[0]?.value) - Number(b[0]?.value);
-    });
-    updateRollingStockCurve(orderedValuesByVelocity);
-  };
+  const spreadsheetCurve = useMemo(() => {
+    const { speeds, max_efforts } = selectedCurve.curve;
+    let filledDataSheet: DataSheetCurve[] = [];
+
+    if (speeds && max_efforts) {
+      filledDataSheet = max_efforts.map((effort, index) => ({
+        speed: speeds[index] !== undefined ? Math.round(msToKmh(speeds[index]!)) : '',
+        // Effort needs to be displayed in kN
+        effort: effort !== undefined ? effort / 1000 : '',
+      }));
+    }
+
+    // Add an empty line for input only if last line is not already empty
+    if (
+      filledDataSheet.length === 0 ||
+      filledDataSheet[filledDataSheet.length - 1].speed !== '' ||
+      filledDataSheet[filledDataSheet.length - 1].effort !== ''
+    ) {
+      filledDataSheet.push({ speed: '', effort: '' });
```

Comment:
With the new librairy, the cell can be nullable (before, if the cell was empty, then its value was `''`). We should take advantage of what the librairy gives us:
- [ ] change every time you test if a value is `''` to test if it is `null`
- [ ] in formatCurve, test whether a cell is null
- [ ] in the type DataSheetCurve, the type should be `number | null` instead of `number | string` for both speed and effort
- [ ] also change the type in EffortCurveForm, `number | null` instead of `number | undefined`
- [ ] in spreadsheetCurve (l94), test whether the speed and effort are null instead of undefined (and set them to null instead of `''`)
- [ ] change the initialization of a new row like this
```suggestion
      filledDataSheet.push({ speed: null, effort: null });
```

Reply:
Done